mockhub.go

```
@@ -0,0 +1,205 @@
+package frontman
+
+// a mock hub used in tests, similar to https://bitbucket.org/cloudradar/debug_hub/
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+type MockHub struct {
+	address string
+}
+
+func NewMockHub(address string) *MockHub {
+	return &MockHub{
+		address: address,
+	}
+}
+
+func (hub *MockHub) URL() string {
+	return "http://" + hub.address
+}
+
+// returns some mocked checks
+func (hub *MockHub) indexHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		hub.getHandler(w, r)
+	case "POST":
+		hub.postHandler(w, r)
+	}
+}
+
+func (hub *MockHub) postHandler(w http.ResponseWriter, r *http.Request) {
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("MockHub: got post data", string(data))
+}
+
+func (hub *MockHub) getHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	query := r.URL.Query()
+	serviceChecksS := query.Get("serviceChecks")
+	webChecksS := query.Get("webChecks")
+
+	serviceChecks, err := strconv.ParseInt(serviceChecksS, 10, 64)
+	if err != nil {
+		serviceChecks = 5
+	}
+
+	webChecks, err := strconv.ParseInt(webChecksS, 10, 64)
+	if err != nil {
+		webChecks = 5
+	}
+
+	log.Println("responding with", serviceChecks, "serviceChecks and", webChecks, "webChecks")
+
+	checks := Input{
+		ServiceChecks: mockServiceChecks(int(serviceChecks)),
+		WebChecks:     mockWebChecks(int(webChecks)),
+	}
+
+	data, _ := json.Marshal(checks)
```

error is ignored? can you enable golangci check for this?